plugins/modules/packaging/os/pacman.py

```diff
@@ -508,7 +536,13 @@ def update_package_db(self):
 
         rc, stdout, stderr = self.m.run_command(cmd, check_rc=False)
 
+        # TODO: check whether the cache was actually updated
+        # Right now, this only seems to be possible by adding `--debug` and looking
```

Comment:
Checking stderr for `response code 2` seems like a decent approach. Alternatively, you could consider the databases up to date iff time conditions are used and are all satisfied:

```
cache_updated = not (0 < stderr.count('using time condition') == stderr.count('file met time condition'))
```

Both the response code and time condition strings are over a decade old at this point so they're somewhat reliable. (They first appeared in pacman 4.0.0, ~October 2011.)

Reply:
There's also the option to compare the output of `pacman -Sl` before and after `-Sy` which avoids having to rely on `--debug` output. It may seem a bit hacky at first but it's comparing the contents of the databases if you think about it. 🐭️

Reply:
I'm currently tending to use `pacman -Sl` instead, it looks less likely to break (even though the debug output has been very stable :) ).

Reply:
I'm currently working on a diff to remove calls to `pacman -Sl` in an effort to make this module usable with yay. (the available package list from AUR is quite long and slow to list.)

I wonder if checking that all meaningful lines from `pacman -Sy` contain "up to date" could be enough to assert that all repos were up to date and nothing was done.

Reply:
something like

```
for l in stdout.splitlines():
  if not l: continue
  if l.startswith(":: "): continue
  if "up to date" not in l:
    changed = True
    break
```

Listing all changed repo is an exercise for someone else :P

Reply:
> I wonder if checking that all meaningful lines from `pacman -Sy` contain "up to date" could be enough to assert that all repos were up to date and nothing was done.

The problem is that `up to date` is only printed when the output goes to a tty. You can for example pipe the output through `cat`, then you can see what the module would get - and that does not contain that information.

Reply:
you're right it doesn't work at all, I didn't test without a tty :-)

Reply:
Still exploring alternatives here: what about looking at the `mtimes` of the actual db file?
Something equivalent to `ls -l $(pacman-conf DbPath)/sync/*db`

Should be cheap and robust enough I think?
if we go that way we should probably add an option to specify the pacman config file too so we can point `pacman-conf` to the right file.

I'm not super fond of this alternative, but so far I'm not fond of the others either: looking past pacman at its databases files, or parsing its debug output or comparing potentially huge package lists.

Reply:
> Still exploring alternatives here: what about looking at the `mtimes` of the actual db file? Something equivalent to `ls -l $(pacman-conf DbPath)/sync/*db`

Will that work any better with yay?

> Should be cheap and robust enough I think? if we go that way we should probably add an option to specify the pacman config file too so we can point `pacman-conf` to the right file.
> 
> I'm not super fond of this alternative, but so far I'm not fond of the others either: looking past pacman at its databases files, or parsing its debug output or comparing potentially huge package lists.

I think we have to stop assuming we can find one way that works great with pacman **and** pacman alternatives. We probably need multiple versions (at least two?) and call these depending on which package manager is actually used.

Reply:
> Will that work any better with yay?

you're right, it wouldn't work at all. All the solutions would be tightly coupled to pacman.

> I think we have to stop assuming we can find one way that works great with pacman **and** pacman alternatives. We probably need multiple versions (at least two?) and call these depending on which package manager is actually used.

you're right here too.

Reply:
I was just thinking back on this, and `yay -S --refresh` just pipes through to `pacman`. In other words, the "look at the db mtimes¨  scheme would work.

Reply:
The main problem with the modification times for the database files is that I don't see how to consistently get hold of the paths to these files. I don't think we can assume that `pacman-conf` is available in the path, and if it would be corresponds to the pacman executable we are given.

Reply:
I see your point.
However adding a `pacman_config` parameter (to allow selection of the pacman config file) would help in any case, so that the file could be changed without resorting to `extra_args` (which would break the inventory functions currently)
As for the path to `pacman-conf`, a default to `/bin/pacman-conf` + a parameter to change it wouldn't be too bad.

But I agree that it is starting to get a bit convoluted.